### DISCUSSION TITLE
initializing miner with ledger wallet owner address fails

### DISCUSSION THREAD

**Discussion by daveydoo123** [filecoin-project/lotus#10919] [Storage Provider] [answered]
I added an f1/secp256k1-ledger address and an f3/bls address to my wallet. Then tried to run the command to initialize a miner using the f1 address as the owner and the f3 as the controller:

```
lotus-miner init --owner=f1z3k6b3u62bpjoptyffwsb7hqcvkvcwuuxqdce6q --worker=f3sh6ewy66uzumgnsq3cthdozpzhsd522agg5ukxcsd5d3wplauynpd6kjmzppjhf6uxftadobolhqciyk3s5a --no-local-storage
```

it fails with this error message:

> 2023-05-23T17:20:50.528-0700	ERROR	main	lotus-miner/init.go:269	Failed to initialize lotus-miner: creating miner failed:
>     main.storageMinerInit
>         /root/lotus/cmd/lotus-miner/init.go:552
>   - pushing createMiner message:
>     main.createStorageMiner
>         /root/lotus/cmd/lotus-miner/init.go:749
>   - failed to sign message: Unexpected data type, addr=f1z3k6b3u62bpjoptyffwsb7hqcvkvcwuuxqdce6q: Unexpected data type

Ledger Filecoin App Version: 0.22.10
Lotus Version: 1.23.0

**Comment by TippyFlitsUK** [comment]
Sorry to hear that you are still having issues with this Davey!
I have found a couple of related issues [here](https://filecoinproject.slack.com/archives/CEGN061C5/p1638816447467500) and [here](https://github.com/filecoin-project/lotus/issues/5127). Can you please make sure that your Ledger is in expert mode and that you are using the latest Ledger firmware and Filecoin app?

https://filecoinproject.slack.com/archives/CPFTWMY7N/p1684875463135779

**Comment by daveydoo123** [comment, marked as answer]
So I did have my Ledger in expert mode and was/am running the latest firmware.

Anyways so after looking at the previous issue you mentioned, I got it to work by first initializing the miner to an f3 address. Then used `lotus-miner actor set-owner --really-do-it` twice to change the owner address from the f3 address to the f1/ledger address. So with that workaround, I'm good now.

However the init using the f1/ledger as the owner address never worked--so that's still an issue.
